.github/workflows/code-analysis.yml

```diff
@@ -0,0 +1,31 @@
+name: PHP Composer
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  php-stan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install packages and run PHP Stan
+      uses: actions/checkout@v2
+    - run: composer install
+    - run: sudo cp vendor/prestashop/php-dev-tools/phpstan/ps-module-extension.neon vendor/prestashop/php-dev-tools/phpstan/extension.neon
+    - run: sudo cp vendor/prestashop/php-dev-tools/templates/phpstan/phpstan.neon phpstan.neon
+    - run: sudo chown -R $USER:$USER phpstan.neon
+    - run: sudo echo '  excludes_analyse:' >> phpstan.neon
+    - run: sudo echo '    - %currentWorkingDirectory%/vendor/*' >> phpstan.neon
+    - run: _PS_ROOT_DIR_=./ php vendor/phpstan/phpstan/phpstan.phar --configuration=phpstan.neon analyse ./
```

Comment:
Looks like a required step is to download the PrestaShop source, this could be done via composer but as it is not available via packagist the process would be very slow, so I would suggest fetching it via wget from https://github.com/PrestaShop/PrestaShop/archive/1.7.7.0.tar.gz and unpacking it as part of the script

Reply:
this is done, we are adding prestashop code as well.